sql/src/main/java/io/crate/protocols/postgres/types/PGArray.java

```diff
@@ -248,18 +249,22 @@ private int decodeUTF8Text(byte[] bytes, int startIdx, int endIdx, List<Object>
     private void addObject(byte[] bytes, int startIdx, int endIdx, List<Object> objects) {
         if (endIdx > startIdx) {
             byte firstValueByte = bytes[startIdx];
-            if (firstValueByte == '"') {
-                List<Byte> innerBytes = new ArrayList<>(endIdx - (startIdx + 1));
-                for (int i = startIdx + 1; i < endIdx; i++) {
-                    if (i < (endIdx - 1) && (char) bytes[i] == '\\' &&
+            if (firstValueByte == 'N') {
+                objects.add(null);
+            } else {
+                if (firstValueByte == '"') {
+                    startIdx++;
+                    endIdx--;
+                }
+                List<Byte> innerBytes = new ArrayList<>(endIdx - (startIdx));
```

Comment:
Maybe as a follow up change this to a `byte[]` array? We know the length up-front and call `Bytes.toArray` later. So by using `byte[]` directly we could avoid some boxing.